apis/v1beta1/config.go

```diff
@@ -225,6 +226,72 @@ func (c *Config) getPortsForComponentKinds(logger logr.Logger, componentKinds ..
 	return ports, nil
 }
 
+// getPortsForComponentKinds gets the ports for the given ComponentKind(s).
+func (c *Config) applyDefaultForComponentKinds(logger logr.Logger, componentKinds ...ComponentKind) error {
+	enabledComponents := c.GetEnabledComponents()
+	for _, componentKind := range componentKinds {
+		var retriever components.ParserRetriever
+		var cfg AnyConfig
+		switch componentKind {
+		case KindReceiver:
+			retriever = receivers.ReceiverFor
+			cfg = c.Receivers
+		case KindExporter:
+			continue
+		case KindProcessor:
+			continue
+		case KindExtension:
+			continue
+		}
+		for componentName := range enabledComponents[componentKind] {
+			parser := retriever(componentName)
+			if newCfg, err := parser.GetDefaultConfig(logger, cfg.Object[componentName]); err != nil {
+				return err
+			} else {
+				// NOTE: The internally used parser returns components.SingleEndpointConfig
+				// as a map value. The following lines normalize this value..
+				got, err := yaml.Marshal(newCfg)
```

Comment:
are we able to use mapstructure here instead?

Reply:
I had that before but did end up with:
```diff
Diff:
--- Expected
+++ Actual
@@ -115,9 +115,11 @@
      (string) (len=4) "otlp": (map[string]interface {}) (len=1) {
-      (string) (len=9) "protocols": (map[string]interface {}) (len=2) {
-       (string) (len=4) "grpc": (map[string]interface {}) (len=1) {
-        (string) (len=8) "endpoint": (string) (len=12) "0.0.0.0:4317"
-       },
-       (string) (len=4) "http": (map[string]interface {}) (len=1) {
-        (string) (len=8) "endpoint": (string) (len=12) "0.0.0.0:4318"
-       }
+      (string) (len=9) "protocols": (map[string]*components.SingleEndpointConfig) (len=2) {
+       (string) (len=4) "grpc": (*components.SingleEndpointConfig)({
+        Endpoint: (string) (len=12) "0.0.0.0:4317",
+        ListenAddress: (string) ""
+       }),
+       (string) (len=4) "http": (*components.SingleEndpointConfig)({
+        Endpoint: (string) (len=12) "0.0.0.0:4318",
+        ListenAddress: (string) ""
+       })
```

Reply:
Well. Thats now obsolete.